File: src/types/diagnosis.ts
export interface BlindSpot {
  tag: string
  severity: string
  description: string
  note_reference: string
  suggestion: string
}

export interface DiagnosisRound {
  role: 'ai' | 'user'
  content: string
  blind_spots: BlindSpot[]
  follow_up?: string
}

export interface DiagnosisReport {
  summary: string
  blind_spots: BlindSpot[]
  overall_level: string
  next_steps: string[]
}

export interface DiagnosisSession {
  session_id: string
  question: string
  user_answer: string
  user_reasoning: string
  note_path: string
  conversation: DiagnosisRound[]
  current_round: number
  max_rounds: number
  final_report?: DiagnosisReport
}

export type DiagnosisEvent =
  | { event: 'initial'; data: DiagnosisRound }
  | { event: 'follow_up'; data: { question: string; blind_spots: BlindSpot[] } }
  | { event: 'report'; data: DiagnosisReport }
  | { event: 'error'; data: { message: string } }

File: src-tauri/src/models/diagnosis.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BlindSpot {
    pub tag: String,
    pub severity: String,
    pub description: String,
    pub note_reference: String,
    pub suggestion: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DiagnosisRound {
    pub role: String,
    pub content: String,
    #[serde(default)]
    pub blind_spots: Vec<BlindSpot>,
    #[serde(default)]
    pub follow_up: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DiagnosisReport {
    pub summary: String,
    pub blind_spots: Vec<BlindSpot>,
    pub overall_level: String,
    pub next_steps: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DiagnosisSession {
    pub session_id: String,
    pub question: String,
    pub user_answer: String,
    pub user_reasoning: String,
    pub note_path: String,
    pub note_content: String,
    pub conversation: Vec<DiagnosisRound>,
    pub current_round: u32,
    pub max_rounds: u32,
    pub final_report: Option<DiagnosisReport>,
}
